engine/src/main/java/com/datatorrent/stram/engine/InputNode.java

```diff
@@ -145,7 +148,7 @@ else if (PROCESSING_MODE == ProcessingMode.EXACTLY_ONCE) {
 
               ContainerStats.OperatorStats stats = new ContainerStats.OperatorStats();
               reportStats(stats, currentWindowId);
-              if(!insideWindow){
+              if(!insideApplicationWindow){
```

Comment:
Metrics are collected at the end of application window. I think Tim has just changed the name of the variable and nothing else has changed. @PramodSSImmaneni please go through older discussions about why Metrics are collected at the end of application window. Also I have an improvement where it phase of metrics collections can be specified.